### DISCUSSION TITLE
Checking for Nones based on an arrays type

### DISCUSSION THREAD

**Discussion by maadcoen** [scikit-hep/awkward#3148] [Q&A] [answered]
I have a collections of arrays of different types, and I want to assure the content does not contain any None values. 
I understood that an array that can contain Nones has this reflected in its type by a question mark. So it's seems like it is not necessary to perform `ak.any(ak.is_none(ARRAY))` on every single array. However, I couldn't find a way to properly inspect the type of an array like this, except for doing `"?" in str(ARRAY.type)`, which feels rather clumsy to me. 
Is there a better way of doing this?

**Comment by jpivarski** [comment, marked as answer]
`ARRAY.type` is a data structure with `content` (single child, another Type) and `contents` (multiple children, a list of Types) that can be navigated. 

It's best to avoid checking for a `"?"` in the string because in more complex cases, the function name `option[...]` is used.
